lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
Actually, I think we should just raise on `nil`. Soon, we'll support reading values from an object with `attr` DSL (a WIP PR is already opened for this feature), so we don't really require a hash.